### PR TITLE
add more details on missing $ref

### PR DESCRIPTION
**Pull request by AlexHolly**
Example Current infos:
https://apitools.dev/swagger-parser/online/

```
openapi: 3.0.2
info:
  description: >-
    This is a sample server Petstore server.
  version: 1.0.0
  title: Swagger Petstore
  termsOfService: 'http://swagger.io/terms/'
  contact:
    email: apiteam@swagger.io
  license:
    name: Apache 2.0
    url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
paths:
  /pet:
    post:
      tags:
        - pet
      parameters:
        - $ref: '#/components/parameters/ThisIsMissing'
components:
  parameters:
    petId:
      name: petId
      in: path
      required: true
      schema:
        type: integer
        format: int64
```

PR will give this data

```
$ref value of: #/paths/~1pet/post/parameters/0 
  is missing target: #/components/parameters/ThisIsMissing
  found: #/components/parameters
```

### CODE REVIEW

**Comment by JamesMessinger** [comment]
Hi @AlexHolly.  Thank you for the PR. It looks like that code should probably go inside the  `MissingPointerError` class instead (with additional data passed as parameters to the constructor).

**Comment by AlexHolly** [comment]
Done.

File: lib/util/errors.js [hunk 1]
```diff
@@ -93,10 +93,14 @@ const UnmatchedResolverError = exports.UnmatchedResolverError = class UnmatchedR
 setErrorName(UnmatchedResolverError);
 
 const MissingPointerError = exports.MissingPointerError = class MissingPointerError extends JSONParserError {
```
Comment (P0lip) [inline lib/util/errors.js:95]: Could you please update the TS definitions as well?

**Review by P0lip — CHANGES_REQUESTED** [review]
Besides the TS definition, a minor test update would be cool.
We presumably already have some tests for `MissingPointerError`, thus adding an additional assertion should be sufficient, no need to write a new case.

**Comment by jonluca** [comment]
This PR looks good - apologies for the long delay here @AlexHolly 

If you rebase this onto latest I'll merge it in.

**Comment by AlexHolly** [comment]
@jonluca I am unable to test it with https://github.com/APIDevTools/swagger-parser looks like it expects .js files. Is there a command to build them?

**Pull request closed without merging**